parlai/core/torch_ranker_agent.py

```diff
@@ -137,6 +142,28 @@ def get_train_preds(self, scores, label_inds, cands, cand_vecs):
             preds = [cands[i][ordering[0]] for i, ordering in enumerate(ranks)]
         return Output(preds)
 
+    def is_valid(self, obs):
+        """Override from TorchAgent."""
+        if not self.opt.get('ignore_bad_candidates', False):
+            return super().is_valid(obs)
+
+        good_ex = 'text_vec' in obs or 'image' in obs
```

Comment:
i think we should also error if not good_ex, if not on robust mode ?

Reply:
and i think  this robustness should be for all torch agents, not just rankers

Reply:
i put it in TRA because if you're not ranking label candidates then this doesn't really matter